pkg/yurtmanager/controller/yurtappoverrider/yurtappoverrider_controller.go

```diff
@@ -71,14 +76,18 @@ var _ reconcile.Reconciler = &ReconcileYurtAppOverrider{}
 // ReconcileYurtAppOverrider reconciles a YurtAppOverrider object
 type ReconcileYurtAppOverrider struct {
 	client.Client
-	Configuration config.YurtAppOverriderControllerConfiguration
+	Configuration  config.YurtAppOverriderControllerConfiguration
+	CacheOverrider *appsv1alpha1.YurtAppOverrider
+	recorder       record.EventRecorder
 }
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(c *appconfig.CompletedConfig, mgr manager.Manager) reconcile.Reconciler {
 	return &ReconcileYurtAppOverrider{
-		Client:        mgr.GetClient(),
-		Configuration: c.ComponentConfig.YurtAppOverriderController,
+		Client:         mgr.GetClient(),
+		Configuration:  c.ComponentConfig.YurtAppOverriderController,
+		CacheOverrider: &appsv1alpha1.YurtAppOverrider{},
```

Comment:
I think we need cache all yurtappoverriders instead of caching only one yurtappoverrider.

Reply:
fix